File: OC/内容层动画/WGOCCoreAnimation/WGOCCoreAnimation/CAEmitterCell粒子动画/霓虹效果/WGNeonAniViewController.m
//
//  WGNeonAniViewController.m
//  WGOCCoreAnimation
//
//  Created by wanggang on 2019/5/23.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGNeonAniViewController.h"

@interface WGNeonAniViewController ()

@property (nonatomic, strong) CAEmitterLayer *emitterLayer;
@property (nonatomic, strong) CAEmitterCell *emitterCell;

@end

@implementation WGNeonAniViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"霓虹效果";
    self.view.backgroundColor = UIColor.blackColor;
    
    //粒子发射器
    self.emitterLayer = [CAEmitterLayer layer];
    //粒子发射位置
    _emitterLayer.position = self.view.center;
    //控制大小
    _emitterLayer.emitterSize = CGSizeMake(2, 2);
    _emitterLayer.renderMode = kCAEmitterLayerBackToFront;
    //控制发射源模式 即形状
    _emitterLayer.emitterMode = kCAEmitterLayerOutline;
    _emitterLayer.emitterShape = kCAEmitterLayerCircle;
    [self.view.layer addSublayer:_emitterLayer];
    
    self.emitterCell = [[CAEmitterCell alloc] init];
    _emitterCell.name = @"nenolight";
    //发射方向
    _emitterCell.emissionLongitude = M_PI;
    //粒子速度
    _emitterCell.velocity = 50;
    //粒子速度范围
    _emitterCell.velocityRange = 50;
    //缩放比例
    _emitterCell.scaleSpeed = -0.2;
    _emitterCell.scale = 0.1;
    _emitterCell.greenSpeed = -0.1;
    _emitterCell.alphaSpeed = -0.2;
    _emitterCell.birthRate = 100;
    _emitterCell.lifetime = 4;
    //颜色
    _emitterCell.color = [UIColor whiteColor].CGColor;
    
    _emitterCell.contents = (__bridge id _Nullable)([UIImage imageNamed:@"nenolight.png"].CGImage);
    
    _emitterLayer.emitterCells = @[_emitterCell];
}



@end

File: Swift/内容层动画/WGCoreAnimation/WGCoreAnimation/CAEmitterCell粒子动画/霓虹效果/WGNeonAniViewController.swift
//
//  WGNeonAniViewController.swift
//  WGCoreAnimation
//
//  Created by wanggang on 2019/4/25.
//  Copyright © 2019 wanggang. All rights reserved.
//

import UIKit

class WGNeonAniViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        title = "霓虹效果"
        view.backgroundColor = UIColor.black
        
        let emitterCell = CAEmitterCell()
        emitterCell.name = "nenolight"
        //发射方向
        emitterCell.emissionLongitude  = CGFloat(Double.pi*2)
        // 粒子速度
        emitterCell.velocity = 50
        // 粒子速度范围
        emitterCell.velocityRange = 50
        // 缩放比例
        emitterCell.scaleSpeed = -0.2
        emitterCell.scale = 0.1
        emitterCell.greenSpeed = -0.1
        emitterCell.redSpeed = -0.2
        emitterCell.blueSpeed = 0.1
        emitterCell.alphaSpeed = -0.2
        emitterCell.birthRate = 100
        emitterCell.lifetime = 4
        emitterCell.color = UIColor.white.cgColor
        emitterCell.contents = UIImage(named:"nenolight.png")?.cgImage
        
        
        let emitterLayer = CAEmitterLayer()
        // 粒子发射位置
        emitterLayer.position = self.view.center
        // 控制粒子大小
        emitterLayer.emitterSize = CGSize(width: 2, height: 2)
        emitterLayer.renderMode = .backToFront
        // 控制发射源模式 即形状
        emitterLayer.emitterMode = .outline
        emitterLayer.emitterShape = .circle
        emitterLayer.emitterCells = [emitterCell]
        
        self.view.layer.addSublayer(emitterLayer)
        
    }


}
